packages/nuxt/src/app/composables/id.ts

```diff
@@ -54,5 +54,5 @@ export function useId (key?: string): string {
   }
 
   // pure client-side ids, avoiding potential collision with server-side ids
-  return key + '_' + nuxtApp._id++
+  return key + SEPARATOR + nuxtApp._id++
```

Comment:
The reason is so that purely client side calls to useId don’t run the risk of being the same id.

I think the fix needs to be different here.

Reply:
I think the problem with `_` vs `-` is that the error message is confusing. 99% people see the difference in 1 character and think it is a bug or something like this. It's not clear from the hydration mismatch error that the ID value was actually generated purely on the client and the server value was ignored because of mutli-root element.

I'm not sure how this problem could be solved. Maybe changing `_` to `_client_` could make it more clear, but of course it's silly "solution". :) Ideally the error message should clearly indicate this, but I'm not sure at the time it is generated, it has access to the information about `useId()` and if it was used at all.